### DISCUSSION TITLE
enable vercel dev to pull environment vars from .env.local

### DISCUSSION THREAD

**Discussion by hassantsyed** [vercel/vercel#9119] [Ideas]
I want to add the functionality of pulling env vars from .env.local to the command 'vercel dev'.
This is to create alignment with NextJS, as it already does this.

It is an inconsistent developer experience having a nextJS frontend being deployed through vercel which pulls from .env.local (for local development) and needing a different environment variable mechanism for other vercel projects.
This also enables continued development in an offline scenario.

I propose adding an additional cli argument.
Perhaps --local (although -l is taken by --listen) or --path [path-to-env]
Ex.
vercel dev --local
- which would populate env vars from .env.local
vercel dev --path [relative-path-to-env]
- which would populate env vars from the relative file path provided.

Changes for populating the env vars would be isolated to here: packages/cli/src/commands/dev/dev.ts
Changes to updating the CLI args would be isolated to here: packages/cli/src/commands/dev/index.ts

I look forward to your feedback and sign off, so I can begin writing this feature.

Thanks!
Hassan Syed

**Comment by amyegan** [comment]
Thanks for sharing this idea @hassantsyed! With the current features, you can upload your local env with [`vercel env add [name] [environment] < [file]`](https://vercel.com/docs/cli/env#extended-usage) and you can pull your env vars into your local environment with [`vercel pull`](https://vercel.com/docs/cli/pull#) to use with the `vercel dev` command. You can also copy/paste to [bulk upload environment variables](https://vercel.com/changelog/bulk-upload-now-available-for-environment-variables).

I'm going to move this to the [Ideas](https://github.com/vercel/vercel/discussions/categories/ideas) category and share it with the team for feedback 🙂

**Reply by hassantsyed** [reply]
Those both require internet connections. It also diverges from the NextJS api.
